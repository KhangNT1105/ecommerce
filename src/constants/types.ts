export type MarketType = {
  countryCode: string;
  countryName: string;
  currencyCode: string;
  currencySymbol: string;
  active: boolean;
};

export type ClientType = {
  clientId: string;
  clientName: string;
  active: boolean;
};

export type PaginationType = {
  totalPages: number;
  totalItems: number;
  hasPrevPage: boolean;
  hasNextPage: boolean;
  prevPage: number | null;
  nextPage: number | undefined;
  size: number | 10;
  page: number | undefined;
};

export type MenuItem = {
  path: string;
  name: string;
  icon: string;
  children: MenuItem[];
};

export type ProductDenominationType = {
  label: string;
  price: number;
};

export interface IProductImage {
  logo: string;
  thumbnail?: string;
  portrait?: string;
  landscape?: string;
}

export interface ICategory {
  id: string;
  name: string;
  parentCategoryId?: string;
  externalCategoryId?: string;
  vendor?: string;
}

export interface IProductItem {
  id: string;
  displayName: string;
  vendor: string;
  productId?: string;
  market?: string;
  productBrand?: string;
  ghostLinkExpiryDays?: number;
  minimumAllowableExpiryDays?: number;
  valueType?: string;
  minPrice?: number;
  maxPrice?: number;
  redemptionType?: string;
  denominations?: string[];
  productDenominations?: ProductDenominationType[];
  ghostLinkInMarket?: number;
  allocated?: number;
  inQueue?: number;
  stockOnHand?: number;
  expiring?: number;
  internalName?: string;
  originalName?: string;
  description?: string;
  externalProductId?: string;
  categoryIds?: string[];
  published?: string;
  outOfStockBehaviour?: string;
  images?: IProductImage;
  brand?: {
    brandLogo?: string;
    brandName?: string;
  };
}

export interface AnyObject {
  [key: string]: any;
}

export interface StringPropertyObject {
  [key: string]: string;
}

export const SortValues = {
  ASC: 'updatedAt.asc',
  DESC: 'updatedAt.desc'
};

export const PortalType = {
  DOMAIN: 'domain',
  SMS: 'sms'
};

export type FnType = (param?: object) => void;
export type FnSubmitFormType = (event: React.FormEvent<HTMLFormElement>) => void;

export type EXPORT_TYPE = 'txt' | 'json' | 'csv' | 'xls' | 'xml';

export type PathParamsWithIDType = {
  id: string;
};
