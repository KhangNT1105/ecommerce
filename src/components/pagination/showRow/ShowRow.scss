.show-row-container {
  display: flex;
  display: -webkit-flex;
  padding: 0 10px;

  p {
    padding: 3px 10px;
    font-size: 15px;
  }

  .dropdown {
    width: 80px;
    top: 0;
    background: none;

    .dropdown-toggle::after {
      font-size: 14px;
    }

    .dropdown-toggle.btn-secondary {
      min-height: 30px;
    }

    .dropdown-menu {
      min-width: 0;
    }
  }
}
