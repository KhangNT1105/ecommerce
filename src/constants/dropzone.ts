import Configs from 'config';

export const UPLOAD_ACCEPT_FILE_TYPE: string = 'image/jpeg, image/jpg, image/png, image/svg, image/bmp, image/gif';
export const UPLOAD_ACCEPT_FILE_VIDEO_TYPE: string =
  'video/mp4, video/webm, video/wmv, video/flv, video/ogv, video/avi, video/mov';
export const FILE_DROPZONE_MIN_SIZE: number = 0;
export const FILE_DROPZONE_MAX_SIZE: number = 10048576;
export const FILE_DROPZONE_VIDEO_MAX_SIZE: number = 1000048576;
export const FILE_SIZE_IN_MB: number = 1;
export const FILE_SIZE_IN_PX: number = 1920;
export const RESOLUTION_MIN_SIZE_LANDSCAPE: number = 0;
export const RESOLUTION_MIN_SIZE_PORTRAIT: number = 0;

export const defaultDropzone = {
  apiURL: Configs.API.UPLOAD_SERVICE,
  files: [],
  autoUpload: true,
  config: {
    accept: UPLOAD_ACCEPT_FILE_TYPE,
    disabled: false,
    noClick: false,
    noKeyboard: false,
    minSize: FILE_DROPZONE_MIN_SIZE,
    maxSize: FILE_DROPZONE_MAX_SIZE
  },
  pushData: {
    error: false,
    msg: 'OK',
    infos: {
      title: 'Hero Banner Desktop and big screens',
      url: 'https://i.pinimg.com/236x/0a/8b/fb/0a8bfb162ef8a917f1e4d4bf5cd08e4c--android-apps-cover-art.jpg',
      size: '12222'
    }
  }
};

export const defaultDropzoneVideo = {
  ...defaultDropzone,
  config: {
    ...defaultDropzone.config,
    accept: UPLOAD_ACCEPT_FILE_VIDEO_TYPE,
    maxSize: FILE_DROPZONE_VIDEO_MAX_SIZE
  }
};

export const defaultDropzonePreviewInit = {
  files: [],
  onFileRemove: () => {},
  onFileUpload: () => {},
  status: false
};
