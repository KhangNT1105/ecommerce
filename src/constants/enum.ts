// TODO: All enum should be listed here
import { TypeOptions, ToastPosition } from 'react-toastify';
import { EXPORT_TYPE } from './types';

export const COLOR = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  SUCCESS: 'success',
  INFO: 'info',
  WARNING: 'warning',
  DANGER: 'danger',
  LINK: 'link'
};

export const SIZE = {
  LARGE: 'lg',
  SMALL: 'sm'
};

export const DURATION = {
  TOAST: 5000,
  TOAST_TRANSITION: 1000
};

interface POSITION_OPTIONS {
  [pos: string]: ToastPosition;
}
export const POSITION: POSITION_OPTIONS = {
  TOP_LEFT: 'top-left',
  TOP_RIGHT: 'top-right',
  TOP_CENTER: 'top-center',
  BOTTOM_LEFT: 'bottom-left',
  BOTTOM_RIGHT: 'bottom-right',
  BOTTOM_CENTER: 'bottom-center'
};

interface TYPE_OPTIONS {
  [type: string]: TypeOptions;
}
export const TOAST_TYPE: TYPE_OPTIONS = {
  SUCCESS: 'success',
  ERROR: 'error'
};

export const DATE_RANGE = {
  TODAY: 'today',
  LAST_7_DAYS: 'last7days',
  LAST_30_DAYS: 'last30days',
  LAST_WEEK: 'lastweek',
  LAST_MONTH: 'lastmonth',
  CUSTOM_DATE: 'customdate'
};

interface FILE_TYPE {
  [pos: string]: EXPORT_TYPE;
}
export const EXPORT_FILE_TYPE: FILE_TYPE = {
  CSV: 'csv',
  XLS: 'xls'
};

export const BREAKPOINTS = {
  EXTRA_LARGE: 1200,
  LARGE: 992,
  MEDIUM: 768,
  SMALL: 576
};

export const DROPZONE_EVENT_TYPE = {
  DROP: 'drop',
  CLICK: 'click'
};

export const ENTITIES = {
  PATH: {
    CONTENT: 'content',
    LICENSES: 'licenses',
    USER_MANAGEMENT: 'user-management',
    SYSTEM_CONFIGURATION: 'system-configuration'
  },
  NAME: {
    SHOWS: 'shows',
    SCHEDULES: 'schedules',
    CHANNELS: 'channels',
    CATALOGS: 'catalogs',
    CATEGORIES: 'categories',
    CASTINGS: 'casts',
    CLIPS: 'clips',
    PLAYLISTS: 'playlists',
    UPDATES: 'updates',
    CONTRACTS: 'contracts',
    LICENSORS: 'licensors',
    SYSTEM_USERS: 'system-users',
    USER_GROUPS: 'user-groups',
    SEASONS: 'seasons',
    CONFIGURATIONS: 'configuration',
    EPISODES: 'episodes',
    PAYMENT_GATEWAYS: 'payment-gateways',
    RISK_ENGINE_RULES_CONFIG: 'risk-engine'
  }
};

export const PUBLIC_PAGES = {
  NAME: {
    WELCOME: '/',
    LOGIN: 'login',
    TAC: 'term-and-conditions',
    ACCESS_DENIED: 'access-denied-403',
    DEMO: '/demo',
    PRODUCTS: 'sanpham',
    PRODUCT: 'sanpham/:id',
    CART: 'cart',
    CHECKOUT: 'checkout'
  }
};

export const LANGUAGES_CODE = {
  VN: 'vn',
  EN: 'en'
};

// This is API's entities from BE
export const API_ENTITIES_NAME = {
  SHOWS: 'shows',
  SCHEDULES: 'schedules',
  CHANNELS: 'channels',
  CATALOGS: 'catalogs',
  CATEGORIES: 'categories',
  CASTINGS: 'casts',
  CLIPS: 'clips',
  PLAYLISTS: 'playlists',
  UPDATES: 'updates',
  CONTRACTS: 'contracts',
  LICENSORS: 'licensors',
  SYSTEM_USERS: 'usergroups', // BE don't have entity's name : users | using same usergroups entity
  USER_GROUPS: 'usergroups',
  SEASONS: 'seasons',
  CONFIGURATIONS: 'configurations',
  EPISODES: 'episodes',
  RISK_ENGINES: 'riskengines'
};

export const PAGE_ACTIONS = {
  CREATE: 'create',
  EDIT: 'edit',
  DELETE: 'delete'
};

export const API_ERROR_CODE = {
  TOKEN_INVALID: '1012',
  TOKEN_EXPIRED: '1011',
  REFRESH_TOKEN_EXPIRED: '1014',
  REFRESH_TOKEN_INVALID: '1015'
};

export const PERMISSION_PREFIX = {
  VIEW: 'VIEW',
  CREATE: 'CREATE',
  EDIT: 'EDIT',
  EDIT_APPROVAL: 'EDIT_APPROVAL',
  EDIT_VIEW_CONTRACT: 'EDIT_VIEW_CONTRACT',
  ACTIVATE: 'ACTIVATE',
  DEACTIVATE: 'DEACTIVATE',
  APPROVE: 'APPROVE',
  REJECT: 'REJECT',
  DELETE: 'DELETE',
  RETRIEVE_BCM: 'RETRIEVE_BCM'
};

export const PAGE_ENTITY = {
  HOME: 'HOME'
};

export const BUTTONS_VISIBILITY = {
  SHOW: 'show'
};
export const PAGE_ACCESSIBILITY = {
  SHOW: 'show'
};

export const CLASS_NAME = {
  READONLY: 'read-only',
  DISABLED: 'disabled'
};

export const SEARCH_BOX_SUBMIT_DELAY = 600;

export const SHOWS_TYPES = {
  MOVIE: 'movie',
  SERIES: 'series',
  LIFE_EVENT: 'life event',
  THEATRICAL_PLAY: 'theatrical play',
  PROGRAM: 'program',
  MATCH: 'match',
  SEASONS: 'seasons',
  CHANNELS: 'channels',
  SCHEDULED: 'scheduled',
  SCHEDULING: 'ready for scheduling'
};

export const IMAGE_TYPES = {
  POSTER: 'poster',
  LANDSCAPE: 'landscape',
  POSTER_CLEAN: 'poster clean',
  LANDSCAPE_CLEAN: 'landscape clean',
  POSTER_WITH_LOGO: 'poster with logo',
  LANDSCAPE_WITH_LOGO: 'landscape with logo',
  LOGO_TITLE: 'logo title',
  HERO_BANNER_DESKTOP: 'hero banner desktop',
  HERO_BANNER_MOBILE: 'hero banner mobile'
};

export const INPUT_FIELD = {
  MAX_LENGTH_NORMAL: 100,
  MAX_LENGTH_URL: 1000
};

export const SORTING_DIRECTION = {
  ASCENDING: 'asc',
  DESCENDING: 'desc'
};
const SECONDS = {
  MINUTE: 60,
  HOUR: 3600,
  DAY: 86400,
  MONTH: 2592000,
  YEAR: 31536000
};
const ENUMS = {
  COLOR,
  SIZE,
  DURATION,
  POSITION,
  TOAST_TYPE,
  DATE_RANGE,
  EXPORT_FILE_TYPE,
  BREAKPOINTS,
  DROPZONE_EVENT_TYPE,
  SECONDS
};

export default ENUMS;
