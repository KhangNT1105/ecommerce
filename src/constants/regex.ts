const REGEX = {
  NUMBER_ONLY: /^[0-9]([.,][0-9])?$/,
  DIGIT_ONLY: /^[0-9]+$/
};

export default REGEX;
