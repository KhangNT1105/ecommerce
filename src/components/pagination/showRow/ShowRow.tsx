import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Dropdown, DropdownMenu, DropdownItem, DropdownToggle } from 'reactstrap';
import { showRowProps } from '../Paginations.d';
import './ShowRow.scss';

const ShowRow: React.FC<showRowProps> = ({ listMaxRows, currentSize, setCurrentSize, handlePagination }) => {
  const [dropdownOpen, setDropdownOpen] = useState(false);
  const { t } = useTranslation();

  const toggle = () => setDropdownOpen((prevState) => !prevState);

  const onDropDownItemClick = (e: any) => {
    const value = e.currentTarget.textContent;
    setCurrentSize(value);
    handlePagination && handlePagination(0, value);
  };

  const listDropDown = listMaxRows.map((item: number, index: number) => {
    return (
      <DropdownItem key={index + item} onClick={onDropDownItemClick}>
        {item}
      </DropdownItem>
    );
  });

  return (
    <div className="show-row-container">
      <p>
        <span>{t('SHOW_ROWS')}</span>
      </p>
      <Dropdown direction="up" isOpen={dropdownOpen} toggle={toggle}>
        <DropdownToggle caret={true}>{currentSize}</DropdownToggle>
        <DropdownMenu>{listDropDown}</DropdownMenu>
      </Dropdown>
    </div>
  );
};

export default ShowRow;
