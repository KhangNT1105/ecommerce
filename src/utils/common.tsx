import React from 'react';
import { Denomination } from 'Props.d';
import { assign, clone, map, reduce, isObjectLike } from 'lodash';
import { validateRequired, validateEmail } from 'utils/validation';
import { ACTIVATE, ENTITY_STATUS_FROM_SERVER, DEACTIVATE, DELETE, APPROVE, REJECT } from '../constants';
import { Genre, SubGenre } from 'stores/ConfigurationStore/configurationType.d';

export const checkRenderDoubleValue = (value1?: string | number, value2?: string | number, value3?: any | '') => {
  let minus = '';
  if (value1 && value2) {
    minus = ' - ';
  }
  return (
    <>
      {value1} {value3 && <b>{value3}</b>}
      {minus}
      {value2}
    </>
  );
};

export type SelectOptionProps = {
  id: string;
  value: string;
  label: string;
};

type DataProps = {
  [key: string]: any;
};

export type SelectOptionKeyMappingProps = {
  key: string;
  value: string;
  label: string;
};

export const renderSelectOptions = (data: DataProps[], keyMapping?: SelectOptionKeyMappingProps) => {
  return (
    <>
      <option value="">Select</option>
      {data.map((item) => {
        if (keyMapping && Object.entries(keyMapping).length > 0) {
          return (
            <option key={item[keyMapping.key]} value={item[keyMapping.value]}>
              {item[keyMapping.label]}
            </option>
          );
        }
        return (
          <option key={item.id} value={item.value}>
            {item.label}
          </option>
        );
      })}
    </>
  );
};

export const checkRenderArray = (items: string[]) => {
  return <span>{items.join(', ')}</span>;
};

export const getListPrices = (denominations: Denomination[]) => {
  const listPrices = denominations.map((item) => item.price);
  return listPrices.sort((a, b) => a - b);
};
export const getPrices = (productDetail: any) => {
  if (productDetail.valueType === 'RANGE') {
    return checkRenderDoubleValue(productDetail.minPrice, productDetail.maxPrice);
  }
  if (productDetail.valueType === 'SLAB') {
    const values = getListPrices(productDetail.denominations);
    return values.join(', ');
  }
};

export interface IPagingModelOption {
  pageIndex: number;
  pageSize: number;
}

export interface IPagingModelResult<T> {
  items: T[];
  hasPrevPage: boolean;
  hasNextPage: boolean;
  prevPage: number | null;
  nextPage: number | null;
  size: number;
  page: number;
  totalItems: number;
  totalPages: number;
}

export function getPaginationResponse<T>(
  pagingModel: IPagingModelOption,
  result: T[],
  totalItems: number
): IPagingModelResult<T> {
  const totalPages = Math.floor((totalItems + pagingModel.pageSize - 1) / pagingModel.pageSize);
  const pagingModelResult: IPagingModelResult<T> = {
    items: result,
    totalItems,
    totalPages,
    hasPrevPage: pagingModel.pageIndex > 0,
    hasNextPage: pagingModel.pageIndex < totalPages - 1,
    prevPage: pagingModel.pageIndex > 0 ? pagingModel.pageIndex - 1 : null,
    nextPage: pagingModel.pageIndex < totalPages - 1 ? pagingModel.pageIndex + 1 : null,
    size: pagingModel.pageSize,
    page: pagingModel.pageIndex
  };
  return pagingModelResult;
}

export function replaceSpecialCharacters(str: string, character: string) {
  return str.replace(/(?:\.(?![^.]+$)|[^\w.])+/g, character);
}

export const classConcat = (classes: string[]): string => {
  if (classes.length) {
    return classes.filter((className) => !!className).join(' ');
  }
  return '';
};

export const recursiveGetFieldName = (obj: object, entityField: string, fieldNameCombine?: string): any => {
  let listFieldConfig: any;
  listFieldConfig = reduce(
    obj,
    (result, value, key) => {
      let cloneResult = clone(result);
      let localFieldName: any = clone(fieldNameCombine);
      if (key === entityField) {
        map(value, (item: any, k) => {
          const getFieldName = `${localFieldName}.${key}.${k}`;

          // check whether any validation
          if (item.validationRules && item.validationRules.length > 0) {
            item.validationRules.map((itemValidate: any, index: number) => {
              if (itemValidate.rule === 'isRequired') {
                item.validationRules[index].validator = validateRequired;
              }
              if (itemValidate.rule === 'isEmail') {
                item.validationRules[index].validator = validateEmail;
              }

              return itemValidate;
            });
          }

          // insert key, value into list field
          return (cloneResult = {
            ...cloneResult,
            [getFieldName]: item
          });
        }); // reset field name

        localFieldName = '';
      } else if (isObjectLike(value)) {
        if (!localFieldName) {
          localFieldName = key;
        } else {
          localFieldName += `.${key}`;
        } // recall recursive fn if not be found the expected entityField

        return assign(cloneResult, recursiveGetFieldName(value, entityField, localFieldName));
      }

      listFieldConfig = {
        ...listFieldConfig,
        ...cloneResult
      };

      return listFieldConfig;
    },
    {}
  );

  return listFieldConfig;
};

export type ObjectFieldType = {
  [field: string]: string | object | string[] | object[] | any;
};

export const trimAllValueWithObjectField = (objField: ObjectFieldType) => {
  if ((!Array.isArray(objField) && typeof objField !== 'object') || objField === null) {
    return objField;
  }
  return Object.keys(objField).reduce(
    (acc: ObjectFieldType, key) => {
      acc[key] =
        typeof objField[key] === 'string'
          ? (objField[key] as string).trim()
          : trimAllValueWithObjectField(objField[key]);
      return acc;
    },
    Array.isArray(objField) ? [] : {}
  );
};

const listShowBtnActivate = [
  ENTITY_STATUS_FROM_SERVER.INACTIVATE,
  ENTITY_STATUS_FROM_SERVER.DRAFT,
  ENTITY_STATUS_FROM_SERVER.UPDATED,
  ENTITY_STATUS_FROM_SERVER.ACTIVATE
];

const listShowBtnSMActivate = [
  ENTITY_STATUS_FROM_SERVER.INACTIVATE,
  ENTITY_STATUS_FROM_SERVER.DRAFT,
  ENTITY_STATUS_FROM_SERVER.UPDATED
];

const listShowBtnSMDelete = [ENTITY_STATUS_FROM_SERVER.DRAFT, ENTITY_STATUS_FROM_SERVER.INACTIVATE];

const listShowBtnDeactivate = [ENTITY_STATUS_FROM_SERVER.ACTIVATE, ENTITY_STATUS_FROM_SERVER.UPDATED];

const listShowBtnDelete = [ENTITY_STATUS_FROM_SERVER.INACTIVATE, ENTITY_STATUS_FROM_SERVER.DRAFT];

export const checkEntityActionNonApproval = (action: string, entityStatus: string) => {
  switch (action) {
    case ACTIVATE:
      return listShowBtnActivate.includes(entityStatus);
    case DEACTIVATE:
      return listShowBtnDeactivate.includes(entityStatus);
    case DELETE:
      return listShowBtnDelete.includes(entityStatus);
    default:
      return true;
  }
};

export const checkSMEntityActionNonApproval = (action: string, entityStatus: string) => {
  switch (action) {
    case ACTIVATE:
      return listShowBtnSMActivate.includes(entityStatus);
    case DEACTIVATE:
      return listShowBtnDeactivate.includes(entityStatus);
    case DELETE:
      return listShowBtnSMDelete.includes(entityStatus);
    default:
      return true;
  }
};

export const getAllSubGenres = (genres: Genre[]) =>
  genres
    .map(({ children }) => children)
    .reduce((currentValue, subGenres) => {
      if (subGenres) {
        return [...currentValue, ...subGenres];
      }
      return currentValue;
    }, []);

export const getAllMicroGenres = (subGenres: SubGenre[]) =>
  subGenres
    .map(({ children }) => children)
    .reduce((currentValue, microGenre) => {
      if (microGenre) {
        return [...currentValue, ...microGenre];
      }
      return currentValue;
    }, []);

const listShowBtnActivateWithApproval = [...listShowBtnActivate, ENTITY_STATUS_FROM_SERVER.REJECT];
const listShowBtnDeactivateWithApproval = [...listShowBtnDeactivate];
const listShowBtnDeleteWithApproval = [...listShowBtnDelete];
const listShowBtnApproveWithApproval = [ENTITY_STATUS_FROM_SERVER.APPROVAL_PENDING];
const listShowBtnRejectWithApproval = [ENTITY_STATUS_FROM_SERVER.APPROVAL_PENDING];

export const checkEntityActionApproval = (action: string, entityStatus: string) => {
  switch (action) {
    case ACTIVATE:
      return listShowBtnActivateWithApproval.includes(entityStatus);
    case DEACTIVATE:
      return listShowBtnDeactivateWithApproval.includes(entityStatus);
    case DELETE:
      return listShowBtnDeleteWithApproval.includes(entityStatus);
    case APPROVE:
      return listShowBtnApproveWithApproval.includes(entityStatus);
    case REJECT:
      return listShowBtnRejectWithApproval.includes(entityStatus);
    default:
      return true;
  }
};

export function createMarkup() {
  return {
    __html: `<div class="zalo-chat-widget" data-oaid="927003371386666481" data-welcome-message="Rất vui khi được hỗ trợ bạn!" data-autopopup="0" data-width="450" data-height="450"></div>
    `
  };
}
