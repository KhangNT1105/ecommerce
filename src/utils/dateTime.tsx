import {
  format,
  parse,
  getTime,
  subDays,
  subMonths,
  startOfWeek,
  endOfWeek,
  startOfMonth,
  endOfMonth,
  startOfToday,
  endOfToday,
  startOfDay,
  endOfDay
} from 'date-fns';

import { DATE_FORMAT, DATE_TIME_FULL_FORMAT } from 'constants/index';

export type DateRangeType = { from: string | null; to: string | null };

export const formatDate = (timestamp: string) => {
  return format(parse(timestamp, 'T', new Date()), DATE_FORMAT);
};

export type DateRangeProps = {
  dateFrom: string | null;
  dateTo: string | null;
};

/**
 * get timestamp from a string
 * @param: value string
 */
export const getTimestamp = (value: string): number => {
  return getTime(parse(value, DATE_FORMAT, new Date()));
};

export const getUTCDateFn = (date: Date) => {
  return new Date(
    date.getUTCFullYear(),
    date.getUTCMonth(),
    date.getUTCDate(),
    date.getUTCHours(),
    date.getUTCMinutes(),
    date.getUTCSeconds()
  );
};

/**
 * Description: combine day from DatePicker and local time then convert to UTC Date
 * @param date: from date picker
 * @return: UTC Date
 */
export const convertDatePickerToUTC = (date: Date) => {
  const today = new Date();
  const combineDateTime = new Date(
    date.getFullYear(),
    date.getMonth(),
    date.getDate(),
    today.getHours(),
    today.getMinutes(),
    today.getSeconds()
  );

  return getUTCDateFn(combineDateTime);
};

/**
 * Description: convert HH:MM:SS string to seconds
 * @param dateString
 * @param second
 */
export const convertTimeStringToMillisecond = (dateString: String) => {
  return (dateString as any).split(':').reduce((acc: any, time: any) => 60 * acc + +time) * 1000;
};

export const convertDateFromToUTC = (date: Date) => {
  const combineDateTime = new Date(date.getFullYear(), date.getMonth(), date.getDate());

  return getUTCDateFn(startOfDay(combineDateTime));
};

export const convertDateToToUTC = (date: Date) => {
  const combineDateTime = new Date(date.getFullYear(), date.getMonth(), date.getDate());

  return getUTCDateFn(endOfDay(combineDateTime));
};

export const getToday = (): DateRangeProps => {
  const dateFrom = getUTCDateFn(startOfToday());
  const dateTo = getUTCDateFn(endOfToday());

  return {
    dateFrom: format(dateFrom, DATE_TIME_FULL_FORMAT),
    dateTo: format(dateTo, DATE_TIME_FULL_FORMAT)
  };
};

export const getLast7days = (): DateRangeProps => {
  const dateTo = format(getUTCDateFn(endOfToday()), DATE_TIME_FULL_FORMAT);
  const dateFrom = format(getUTCDateFn(subDays(startOfToday(), 7)), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLast30days = (): DateRangeProps => {
  const dateTo = format(getUTCDateFn(endOfToday()), DATE_TIME_FULL_FORMAT);
  const dateFrom = format(getUTCDateFn(subDays(startOfToday(), 30)), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLastWeek = (): DateRangeProps => {
  const dateFrom = format(getUTCDateFn(startOfWeek(subDays(startOfToday(), 7))), DATE_TIME_FULL_FORMAT);
  const dateTo = format(getUTCDateFn(endOfWeek(subDays(endOfToday(), 7))), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const getLastMonth = (): DateRangeProps => {
  const dateFrom = format(getUTCDateFn(startOfMonth(subMonths(startOfToday(), 1))), DATE_TIME_FULL_FORMAT);
  const dateTo = format(getUTCDateFn(endOfMonth(subMonths(endOfToday(), 1))), DATE_TIME_FULL_FORMAT);

  return { dateFrom, dateTo };
};

export const convertMillisecondToHHMMSS = (milliseconds: number) => {
  if (!milliseconds) return '';
  const date = new Date(milliseconds);
  const hh = date.getUTCHours();
  const mm = date.getUTCMinutes();
  const ss = date.getSeconds();
  const hours = hh < 10 ? `0${hh}` : `${hh}`;
  const minutes = mm < 10 ? `0${mm}` : `${mm}`;
  const seconds = ss < 10 ? `0${ss}` : `${ss}`;

  return `${hours}:${minutes}:${seconds}`;
};
