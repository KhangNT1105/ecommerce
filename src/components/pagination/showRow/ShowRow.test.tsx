import React from 'react';
import { shallow } from 'enzyme';
import ShowRow from './ShowRow';

import { LIST_MAX_ROWS } from '../../../constants/index';

import { testSnapshots } from '../../../utils/test';
const e = {
  currentTarget: {
    textContent: '50'
  }
};
const defaultProps = {
  listMaxRows: LIST_MAX_ROWS,
  currentSize: 25,
  setCurrentSize: jest.fn(),
  handlePagination: jest.fn()
};
describe('<ShowRow />', () => {
  testSnapshots(ShowRow, [
    {
      props: { ...defaultProps },
      description: 'render snapshot correctly'
    }
  ]);
});

describe('Test Logic', () => {
  // test('should update current size when change dropDown', () => {
  //   const wrapper = shallow(<ShowRow {...defaultProps} />);
  //   wrapper.find('DropdownItem').first().simulate('click', e);
  //   expect(defaultProps.setCurrentSize).toHaveBeenCalledWith(e.currentTarget.textContent);
  // });
  // test('should call api when selected row and return page 1', () => {
  //   const wrapper = shallow(<ShowRow {...defaultProps} />);
  //   wrapper.find('DropdownItem').first().simulate('click', e);
  //   expect(defaultProps.handlePagination).toHaveBeenCalledWith(0, e.currentTarget.textContent);
  // });
});
