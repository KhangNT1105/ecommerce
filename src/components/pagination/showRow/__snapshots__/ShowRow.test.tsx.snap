// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowRow /> snapshots render snapshot correctly 1`] = `
<div
  className="show-row-container"
>
  <p>
    <span>
      SHOW_ROWS
    </span>
  </p>
  <div
    className="dropup dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-label={null}
      className="dropdown-toggle btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      25
    </button>
    <div
      aria-hidden={true}
      className="dropdown-menu"
      role="menu"
      tabIndex="-1"
    >
      <button
        className="dropdown-item"
        onClick={[Function]}
        role="menuitem"
        tabIndex="0"
        type="button"
      >
        15
      </button>
      <button
        className="dropdown-item"
        onClick={[Function]}
        role="menuitem"
        tabIndex="0"
        type="button"
      >
        18
      </button>
      <button
        className="dropdown-item"
        onClick={[Function]}
        role="menuitem"
        tabIndex="0"
        type="button"
      >
        21
      </button>
    </div>
  </div>
</div>
`;
